airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_stream.py

```diff
@@ -39,11 +40,11 @@ class DeclarativeStream(Stream):
     _name: str = field(init=False, repr=False, default="")
     _primary_key: str = field(init=False, repr=False, default="")
     _schema_loader: SchemaLoader = field(init=False, repr=False, default=None)
-    stream_cursor_field: Optional[Union[List[str], str]] = None
+    stream_cursor_field: Optional[Union[InterpolatedString, str]] = None
```

Comment:
This was an interesting gap. We used to hard code this to not support interpolation just a simple or nested cursor field. But this didn't align with the slicer's cursor_field. No connectors used a nested cursor field so far, but we can always extend this out as new functionality if we need it.